sos/plugins/etcd.py

```diff
@@ -23,15 +23,46 @@ class etcd(Plugin, RedHatPlugin):
     """etcd plugin
     """
 
+    packages = ('etcd',)
+
     def setup(self):
-        self.add_copy_spec("/etc/etcd")
+        self.address = self.get_address()
+        self.cmd = 'etcdctl '
+
+        self.add_copy_spec('/etc/etcd')
+
+        subcmds = ['--version',
+                   'member list',
+                   'cluster-health',
+                   'ls --recursive',
+                   ]
+
+        for sub in subcmds:
+            self.add_cmd_output("%s %s" % (self.cmd, sub))
+
+        api_urls = ['/v2/stats/leader',
+                    '/v2/stats/self',
+                    '/v2/stats/store',
+                    ]
+
+        for url in api_urls:
+            self.add_cmd_output("curl -s %s%s" % (self.address, url))
```

Comment:
Ditto list formation rather than driving `add_cmd_putput()` in a loop. This is why these calls all accept a list as argument.